library/ssl_tls13_generic.c

```diff
@@ -215,6 +217,163 @@ int mbedtls_ssl_tls13_write_sig_alg_ext( mbedtls_ssl_context *ssl,
 
 #endif /* MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED */
 
+/*
+ *
+ * STATE HANDLING: Incoming Finished
+ *
+ */
+
+/*
+ * Overview
+ */
+
+/* Main entry point: orchestrates the other functions */
+int mbedtls_ssl_tls1_3_finished_in_process( mbedtls_ssl_context* ssl );
```

Comment:
OK, I suggest something like ..._process_server_finished and ..._write_client_finished - i.e. <verb>_<message_name>. This is the naming scheme used in e.g. ssl_tls13_client.c

Reply:
@ronald-cron-arm do you have a view on this one?

Reply:
The code in ssl_tls13_generic.c is common code, it can be called both in client and server side.  Currently, we only work on client side, and we change the name from finished_in and finished_out  to server_fnished and client_finished, that's ok. But once we want to merge server side code, you will find that it isn't very proper. One example for finished_in:  for the server side,  we have to treat the finished_in as MBEDTLS_SSL_CLIENT_FINISHED, once we rename it to server_fnished, it will be very strange.

Reply:
OK, if it is generic for both server and client then yes agreed we should not include "server" or "client" in the name.

But currently, mbedtls_ssl_tls13_process_finished_in includes debug lines that say "=> parse server finished_in_process". So either it is specifically for processing server finished (not client finished), or the debug line is wrong?

Reply:
https://github.com/ARMmbed/mbedtls/pull/4952#discussion_r726776117 is not addressed. 
I think the debug lines and `mbedtls_ssl_set_state` should be called in `ssl_tls13_client.c` . And most case , I think we can move `*_postprocess` to `ssl_tls13_client.c` . That's more clean to me.

Reply:
Yeap, the debug line is wrong. I will change that.

Reply:
Hi Jerry,  Have you checked the latest code? mbedtls_ssl_set_state  have been there in ssl_tls13_client.c.

Reply:
Sorry, miss the code different.

Reply:
In #4965, I've suggested some function name changes (just today). In the scope of the MVP, that's #4965 we are interested in (client side only), thus I would prefer to merge #4965 first and to decide on the function names in #4965.

Reply:
Hi @ronald-cron-arm , Cause #4965 has some dependencies on this PR, maybe we should merge this PR firstly. Anyway, I will try to sync the comments from #4965 to this PR.

Reply:
Strictly speaking for the MVP we only have to merge #4965. I am ok to merge this one as well as I expect very few additional code to support the server side as well, but I would prefer to merge #4965 first. Could you focus on #4965 and once merged rebase this one on top of it ?

Reply:
Sorry, all of the two PRs are for client side, I just seperate them into two parts.  What the two PRs do is reading server finished in from server side  and send out the client finished out to the server side.

Reply:
From the comments, the function names have been changed to mbedtls_ssl_tls13_process_finished_message, ssl_tls13_preprocess_finished_message(), ssl_tls13_parse_finished_message(), ssl_tls13_postprocess_finished_message(), ssl_tls13_postprocess_server_finished_message(), so how about this conversation? shall we close it?